### PR TITLE
adding function to import vector from matrix market 

### PR DESCRIPTION
**Pull request by RiccardoRossi**
and fixing the function gather values

### CODE REVIEW

File: applications/TrilinosApplication/trilinos_space.h [hunk 1]
```diff
@@ -574,6 +575,46 @@ class TrilinosSpace
...
+        std::vector<double> values(NumMyRows);
+        pv->ExtractCopy(values.data());
+
+        VectorPointerType final_vector = Kratos::make_shared<VectorType>(MyMap);
+        int ierr = final_vector->ReplaceGlobalValues(gids.size(),gids.data(), values.data());
+        if(ierr != 0) KRATOS_THROW_ERROR(std::logic_error,"Epetra failure found","");
+
+        final_vector->GlobalAssemble();
+
+        //defining the importer class
```
Comment (loumalouomega) [inline applications/TrilinosApplication/trilinos_space.h]: Can you clean this ununsed code?

File: applications/TrilinosApplication/trilinos_space.h [hunk 2]
```diff
@@ -574,6 +575,46 @@ class TrilinosSpace
...
+    {
+        KRATOS_TRY
+
+        Epetra_Map MyMap(n, 0, Comm);
+        Epetra_Vector* pv = nullptr;
+
+        int error_code = EpetraExt::MatrixMarketFileToVector(FileName.c_str(), MyMap, pv);
+
+        if(error_code != 0)
+            KRATOS_ERROR << "error thrown while reading Matrix Market Vector file "<<FileName
```
Comment (loumalouomega) [inline applications/TrilinosApplication/trilinos_space.h]: KRATOS_ERROR_IF

**Review by loumalouomega — COMMENTED** [review]
Minor comments

File: applications/TrilinosApplication/custom_python/add_trilinos_space_to_python.cpp [hunk 3]
```diff
@@ -287,7 +300,10 @@ void  AddBasicOperations(pybind11::module& m)
     .def("CreateEmptyMatrixPointer", CreateEmptyMatrixPointer)
     .def("CreateEmptyVectorPointer", CreateEmptyVectorPointer)
     .def("ReadMatrixMarketMatrix", ReadMatrixMarketMatrix)
+    .def("ReadMatrixMarketVector", ReadMatrixMarketVector)
     .def("SetValue", SetValue)
+    //.def("GetValue", GetValue) //deliberately commented out. Only works for local Ids
```
Comment (philbucher) [inline applications/TrilinosApplication/custom_python/add_trilinos_space_to_python.cpp:305]: could we implement a check for the Id if it is local to not lose this functionality?

File: applications/TrilinosApplication/trilinos_space.h [hunk 4]
```diff
@@ -574,6 +575,46 @@ class TrilinosSpace
         KRATOS_CATCH("");
     }
 
+    VectorPointerType ReadMatrixMarketVector(const std::string FileName,Epetra_MpiComm& Comm, int n)
```
Comment (philbucher) [inline applications/TrilinosApplication/trilinos_space.h]:
```suggestion
    VectorPointerType ReadMatrixMarketVector(const std::string& rFileName,Epetra_MpiComm& Comm, int n)
```

File: applications/TrilinosApplication/custom_python/add_trilinos_space_to_python.cpp (hunk 3 above)
Comment (RiccardoRossi) [reply applications/TrilinosApplication/custom_python/add_trilinos_space_to_python.cpp:305]: i did not remove this, it was simply not exported. Note that the input is a global id, and if it is not local an error is thrown, so i would not export this... instead you should use the GatherValues function which i now verified to work

File: applications/TrilinosApplication/trilinos_space.h (hunk 1 above)
Comment (RiccardoRossi) [reply applications/TrilinosApplication/trilinos_space.h]: done

File: applications/TrilinosApplication/trilinos_space.h (hunk 2 above)
Comment (RiccardoRossi) [reply applications/TrilinosApplication/trilinos_space.h]: done

File: applications/TrilinosApplication/trilinos_space.h (hunk 4 above)
Comment (RiccardoRossi) [reply applications/TrilinosApplication/trilinos_space.h]: done

**Comment by philbucher** [comment]
@RiccardoRossi you are not following the style guide in many places
In order to avoid astethic changes in future PRs, could you please update the namings? I can help you if you want

**Comment by RiccardoRossi** [comment]
@philbucher i corrected the obvious stuff with camel and not camel case, nevertheless please feel free to modify it as you consider. 

once we converged, i would also pass astyle to the file prior to merging

**Review by philbucher — APPROVED** [review]
good from my side if it works :)

also +1 for applying astyle

**Pull request merged**